simulators/portal/Dockerfile

```diff
@@ -4,6 +4,10 @@ FROM rust:1.81.0 AS builder
 RUN USER=root cargo new --bin portal
 WORKDIR /portal
 
+# declare build args for first stage
+ARG PORTAL_CONSENSUS_URL
```

Comment:
ARG variables in Docker are scoped to the stage they appear in (or any subsequent stages that inherit from that stage). If you define them in a builder stage but never reference them there (e.g., they aren’t used in a RUN command), they do nothing.

if they’re truly only needed in the final image, you can omit them in the builder stage altogether.